packages/venia-ui/lib/components/MagentoRoute/magentoRoute.js

```diff
@@ -14,9 +14,19 @@ const MESSAGES = new Map()
 
 const MagentoRoute = () => {
     const talonProps = useMagentoRoute();
-    const { component: RootComponent, id, isLoading, routeError } = talonProps;
+    const {
+        component: RootComponent,
+        id,
+        isLoading,
+        routeError,
+        redirectCode,
+        relativeUrl
+    } = talonProps;
 
-    if (isLoading) {
+    if (redirectCode === 301 || redirectCode === 302) {
+        window.location = relativeUrl;
```

Comment:
I'm wondering if this logic shouldn't be in `useMagentoRoute`. In other words, have it return the details for the component that matches `relativeUrl` if the redirect condition is satisfied.

I'm thinking it would save us a redirect on the front-end (page load time) in the case of redirects.

Reply:
Use `history.replace` here instead of directly assigning to `window.location` for consistency

Reply:
Consider putting this in a `useEffect` in the talon

Reply:
Resolved by https://github.com/magento/pwa-studio/pull/2504/commits/c56798d3a174fd0d2ffeaada12f263fd3c827981.